### PR TITLE
RC support by external policy part1

### PR DESCRIPTION
**Pull request by AByzhynar**
Added RC support by external policy component

### CODE REVIEW

**Comment by AByzhynar** [comment]
@dev-gh @LuxoftAKutsan @AKalinich-Luxoft @LitvinenkoIra Please review

File: src/components/policy/policy_external/include/policy/policy_manager_impl.h [hunk 1]
```diff
@@ -174,6 +179,45 @@ class PolicyManagerImpl : public PolicyManager {
...
+                                const std::string& rpc,
+                                const RemoteControlParams& params);
+  void CheckPTUUpdatesChange(
+      const utils::SharedPtr<policy_table::Table> pt_update,
+      const utils::SharedPtr<policy_table::Table> snapshot);
+  bool CheckPTURemoteCtrlChange(
+      const utils::SharedPtr<policy_table::Table> pt_update,
+      const utils::SharedPtr<policy_table::Table> snapshot);
+
+  void CheckPTUZonesChange(
```
Comment (dev-gh) [inline src/components/policy/policy_external/include/policy/policy_manager_impl.h]: @AByzhynar 
There should not be any zones

Comment (AByzhynar) [reply src/components/policy/policy_external/include/policy/policy_manager_impl.h]: @dev-gh Thank you. Done. Commit changed and updated. See 8472267534f313c093864097fdb8de000ac93fa1

File: src/components/include/test/policy/policy_external/policy/mock_policy_manager.h [hunk 2]
```diff
@@ -148,6 +148,51 @@ class MockPolicyManager : public PolicyManager {
...
           const std::string& application_id,
           const rpc::policy_table_interface_base::AppHmiTypes& hmi_types));
+#ifdef SDL_REMOTE_CONTROL
+  MOCK_METHOD2(SetDefaultHmiTypes,
+               void(const std::string& application_id,
+                    const std::vector<int>& hmi_types));
+  MOCK_METHOD2(GetHMITypes,
+               bool(const std::string& application_id,
+                    std::vector<int>* app_types));
+  MOCK_METHOD6(CheckAccess,
```
Comment (AKalinich-Luxoft) [inline src/components/include/test/policy/policy_external/policy/mock_policy_manager.h]: @AByzhynar probably you should update this mock. For example here `MOCK_METHOD5`

File: src/components/include/test/policy/policy_external/policy/mock_policy_manager.h [hunk 3]
```diff
@@ -148,6 +148,51 @@ class MockPolicyManager : public PolicyManager {
...
+               TypeAccess(const PTString& device_id,
+                          const PTString& app_id,
+                          const PTString& module,
+                          const PTString& rpc,
+                          const RemoteControlParams& params));
+  MOCK_METHOD2(CheckModule,
+               bool(const PTString& app_id, const PTString& module));
+  MOCK_METHOD5(SetAccess,
```
Comment (LitvinenkoIra) [inline src/components/include/test/policy/policy_external/policy/mock_policy_manager.h]: @AByzhynar `MOCK_METHOD4`

**Review by dev-gh — APPROVED** [review]

File: src/components/include/test/policy/policy_external/policy/mock_policy_manager.h (hunk 3 above)
Comment (AByzhynar) [reply src/components/include/test/policy/policy_external/policy/mock_policy_manager.h]: @LitvinenkoIra Thank you. Done in c0388440a9f226c54334640621a4897b915e835e

File: src/components/include/test/policy/policy_external/policy/mock_policy_manager.h (hunk 2 above)
Comment (AByzhynar) [reply src/components/include/test/policy/policy_external/policy/mock_policy_manager.h]: @AKalinich-Luxoft Thank you. Done in c0388440a9f226c54334640621a4897b915e835e

**Pull request merged**